src/backend/chat/commands/builtin/quotes.ts

```diff
@@ -278,6 +285,13 @@ export const QuotesManagementSystemCommand: SystemCommand<{
                     const channelData = await TwitchApi.channels.getChannelInformation();
 
                     const currentGameName = channelData && channelData.gameName ? channelData.gameName : "Unknown game";
+                    
+                    // //If no @ is included in the originator arg, set to @streamerName and the rest is the quote
```

Comment:
This check should be over Line 280, as `!quote add AHHHHHHHHHH` would fail out before it reaches this point (only 2 args)

Reply:
Good point, silly me none of my testing was a single word quote.

Reply:
I've got home and applied a fix, but I'm considering re-raising this PR as I'm seeing something strange with my local build. 

Probably down to my ignorance as this is my first time working with JavaScript/TypeScript.

Reply:
Kinda a nitpick, but could you move the following so they'll still be under the if statements?

```js
const channelData = await TwitchApi.channels.getChannelInformation();

const currentGameName = channelData && channelData.gameName ? channelData.gameName : "Unknown game";
```

If this is above, whether or not the command was valid, it's going to make an API call to get channel information, which will cause unnecessary delay if the command was used incorrectly.

Other than that, looks good to me! :D

Reply:
Maybe I'm missing something more elegant here? The new section (quoted below) depends on already having fetched `channelData.displayname`, so the API call can't come _after_ the if statements as far as I can see?

```typescript
if (commandOptions.defaultStreamerAttribution && !args[1].includes("@")) {
    args.splice(1,0,`@${channelData.displayName}`)
}
```

Reply:
Oh, hmm, didn't catch that

In that case, I'd say go this route

```js
const shouldInsertUsername = commandOptions.defaultStreamerAttribution && !args[1]?.includes("@");
const expectedArgs = shouldInsertUsername
  ? 2
  : 3;
if (args.length < expectedArgs) {
  await twitchChat.sendChatMessage(`Please provide some quote text!`);
  return resolve();
}

if (shouldInsertUsername) {
  // your code
}
```

Reply:
May not need the nullish check on `args[1]`, forgetting if the linter will yell at that

Reply:
maybe to be safe

```js
const shouldInsertUsername = !args.length || (commandOptions.defaultStreamerAttribution && !args[1].includes("@"));
```

Reply:
> maybe to be safe
> 
> ```js
> const shouldInsertUsername = args.length === 1 || (commandOptions.defaultStreamerAttribution && !args[1].includes("@"));
> ```

Sorry @Oceanity I've been flat out and unable to try this suggestion out. Assuming I've understood it of course, I hope I've addressed it in b6e4ce273a1b8545f63b3e9b9d0cd40a5ae39cdf?

Reply:
That branch looks like it still has the method in the same order. So, the idea is to move the invalid state above the API call.

Like, currently how you have it is:
- API call to Twitch to get `channelData`
- Check if the command provided was actually valid
- Fail out if not, so API call is _always_ made, even when in a fail state

Ideally it should be:
- Check if the command provided was actually valid
- Fail out if not
- API call to Twitch to get `channelData`

I should have probably edited all my notes together instead of tacking on multiple comments because that probably made things unnecessarily confusing 😛

So here's my full proposal:

```js
case "add": {
  // Get info on whether or not command is correct
  // If we need to add username, one less arg required, but don't poll API yet
  // `args.length === 1` is to prevent the check on `args[1].includes` throwing
  // Exception: can't access method 'includes' on undefined
  // `!quote add @Oceanity blablabla` needs minimum 3 args
  // `!quote add blablabla` needs minimum 2 args 
  const shouldInsertUsername = args.length === 1 || commandOptions.defaultStreamerAttribution && !args[1].includes("@");
  const expectedArgs = shouldInsertUsername
    ? 2
    : 3;
    
  // If args count not met, fail out after only initializing 2 bools and 0 API calls
  // (could even remove `expectedArgs` and inline that here, but this is more readable)
  if (args.length < expectedArgs) {
    await twitchChat.sendChatMessage(`Please provide some quote text!`);
    return resolve();
  }

  // Now that it is for sure a valid quote add, start getting the data from the API
  const channelData = await TwitchApi.channels.getChannelInformation();
  const currentGameName = channelData && channelData.gameName ? channelData.gameName : "Unknown game";
  
  if (shouldInsertUsername) {
    args.splice(1,0,`@${channelData.displayName}`);
  }
  
  // rest of Add Quote logic
}
```

Hopefully this is a bit easier to follow! 😁

Reply:
@Oceanity I definitely _did_ get a confused by the stream of consciousness style 😅 Thanks very much for laying it out again like that 🙇‍♂️

Reply:
@Oceanity I'm pretty confident that I've actually addressed your request in a5b0c294d2874d08fbff06d9f42e761c0ca0524a. 

I did tweak it a little so as not to circumvent the opt-in requirement @ebiggz requested; we don't want to jump to inserting the streamer name if the Option has not been selected in the settings.